tests/unit/test_gce_clients.py

```diff
@@ -0,0 +1,344 @@
+
+            results = await crm_client.list_all_active_projects()
+
+        expected_rsp = crm_one_page_rsp['projects']
+        expected_rsp.append(page2['projects'].pop(0))
+        assert results == expected_rsp
+
+
+class TestGCPClient:
+    @pytest.fixture
+    def compute_base_url(self):
+        return ('https://www.googleapis.com/compute/v1/projects/'
+                'test-project/aggregated/instances?')
+
+    @pytest.fixture
+    def compute_rsp(self):
+        return {
+            'kind': 'compute#instanceAggregatedList',
+            'id': 'd34dbeef',
+            # Simplified Instance resource
+            'items': {
+                'us-west1-z': {
+                    'instances': [{
+                        'id': '1',
+                        'creationTimestamp': '2018-01-01 00:00:00.0000',
+                        'name': 'instance-1',
+                        'description': 'guc3-instance-1-54kj',
+                        'tags': {
+                            'items': ['some-tag'],
+                            'fingerprint': ''
+                        },
+                        'machineType': 'n1-standard-1',
+                        'status': 'RUNNING',
+                        'statusMessage': 'RUNNING',
+                        'zone': 'us-west9-z',
+                        'canIpForward': False,
+                        'networkInterfaces': [{
+                            'network': 'network/url/string',
+                            'subnetwork': 'subnetwork/url/string',
+                            'networkIP': '192.168.0.1',
+                            'name': 'test-network',
+                            'accessConfigs': [{
+                                'type': 'ONE_TO_ONE_NAT',
+                                'name': 'EXTERNAL NAT',
+                                'natIP': '1.1.1.1',
+                                'kind': 'compute#accessConfig'
+                            }],
+                        }],
+                        'metadata': {
+                            'items': [
+                                {'key': 'default', 'value': 'true'}
+                            ],
+                        },
+                    }],
+                    'warning': {
+                        'warning': 'object'
+                    }
+                }
+            },
+            'nextPageToken':
+            None
+        }
+
+    @pytest.mark.asyncio
+    async def test_list_instances_from_one_page(
+            self, compute_rsp, compute_base_url, get_gce_client):
+        gcp_client = get_gce_client(gce_clients.GCPClient)
+        with aioresponses() as m:
+            m.get(
+                compute_base_url + 'maxResults=10&filter=zone eq us-west9-z',
+                payload=compute_rsp)
+
+            results = await gcp_client.list_instances(
+                'test-project',
+                page_size=10,
+                instance_filter='zone eq us-west9-z')
+
+            expected_results = [{
```

Comment:
I don't think this needs to be within the `with aioresponses() as m` context, and can be outside.